.github/workflows/lint-codeql.yaml

```diff
@@ -15,7 +15,13 @@ on:
 permissions: read-all
 
 jobs:
+  check_repo:
+    name: Only run on the cilium/cilium repo
+    if: ${{ github.repository_owner == 'cilium' && github.repository_name == 'cilium' }}
```

Comment:
It would be enough to merge that condition with the existing condition in `analyze`.

Reply:
Oops, I thought you shared the link on purpose to get early reviews.

Reply:
No worries, I was doing a bunch of searching to find the best way to achieve this and then got pulled into a meeting just after pushing a first guess.

Reply:
Pushing a link was so that people would know I was looking at the problem so there wouldn't be duplicate work.